src/EnvironmentVariables.php

```diff
@@ -0,0 +1,163 @@
+<?php
+
+namespace Dotenv;
+
+use ArrayAccess;
+
+/**
+ * @author Nikola Posa <posa.nikola@gmail.com>
+ */
+class EnvironmentVariables implements ArrayAccess
+{
+    /**
+     * @var bool
+     */
+    protected $immutable;
+
+    protected function __construct($immutable)
+    {
+        $this->immutable = $immutable;
+    }
+
+    /**
+     * Creates EnvironmentVariables instance.
+     *
+     * @return EnvironmentVariables
+     */
+    public static function create()
+    {
+        return new self(false);
```

Comment:
If we go down the "final" path, it would be good to do protected ->private modifications also, but since there's no specific interface that describes EnvironmentVariables implementations, I think there's no point in making this class final. 

So I'll fix this method to support late static binding as you suggested.